PHPCompatibility/Tests/Constants/TemporarilyRemovedConstantsUnitTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2023 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\Constants;
+
+use PHPCompatibility\Tests\BaseSniffTest;
+
+/**
+ * Test the ReaddedConstants sniff.
+ *
+ * @group  constants
+ * @covers \PHPCompatibility\Sniffs\Constants\TemporarilyRemovedConstantsSniff
+ * @since  10.0.0
+ */
+class TemporarilyRemovedConstantsUnitTest extends BaseSniffTest
+{
+
+    /**
+     * Test re-added constant sniff
+     *
+     * @param string $constantName Name of the PHP constant.
+     * @param int[]  $lines        Line numbers in the test file which apply to this constant.
+     * @param string $phpVersion   PHP version (or range) to test with.
+     * @param bool   $shouldError  If we expect to get an error or not from the sniff.
+     * @param string $removedIn    PHP version this feature was removed (for the error string).
+     * @param string $addedIn      PHP version this feature was re-addded (for the error string).
+     *
+     * @return       void
+     * @dataProvider dataReaddedConstants
+     * @covers       ::process()
+     */
+    public function testReaddedConstant($constantName, $lines, $phpVersion, $shouldError, $removedIn, $addedIn)
+    {
+        $error = "The constant \"$constantName\" was removed in PHP $removedIn and re-added in PHP $addedIn.";
+
+        $file = $this->sniffFile(__FILE__, $phpVersion);
+        foreach ($lines as $line) {
+            if ($shouldError) {
+                $this->assertError($file, $line, $error);
+            } else {
+                $this->assertNoViolation($file, $line);
+            }
+        }
+    }
+
+    /**
+     * Data provider.
+     *
+     * @see testReaddedConstant()
+     *
+     * @return array
+     */
+    public static function dataReaddedConstants()
+    {
+        return [
```

Comment:
I've removed much of the redundancy in the data provider. Please let me know if you'd like to see this reduced further.

Reply:
I think it can be reduced even further. I also think the test should maybe be split into two tests:
One which tests that the error message is thrown correctly.
A second which tests that no error message is thrown when the `testVersion` doesn't warrant it.

IMO, the first could then test against `7.0`, `7.4`, `-7.2`, `7.2-` and `5.0-8.3`.
For the second, `5.6`, `7.4`, `-5.6`, `7.4-` would suffice (or even just the first two or the last two).

FYI: for every `testVersion` used in the tests, the file is being reprocessed as "config" values can not be changed on the fly.
This reprocessing can be slow, so yes, the tests should be comprehensive, but we tend to avoid redundancy when it involves a lot of testVersions to keep things workable. Having said, that, I have recently done some work to make the tests faster (#1545), so my response may be rooted still in the situation from before that change...